### PR TITLE
Add topic for "cst"

### PR DESCRIPTION
**Pull request by shaedrich**
### Please confirm this pull request meets the following requirements:

- [x] I followed the contributing guidelines: <https://github.com/github/explore/blob/main/CONTRIBUTING.md>.
- [x] I am not the sole author or employee of a company who created the topic or collection I'm changing.

### Which change are you proposing?

  - [ ] Suggesting edits to an existing topic or collection
  - [x] Curating a new topic or collection
  - [ ] Something that does not neatly fit into the binary options above

---

### Editing an existing topic or collection

I'm suggesting these edits to an existing topic or collection:
- [ ] Image (and my file is `*.png`, square, dimensions 288x288, size <= 75 kB)
- [x] Content (and my changes are in `index.md`)

Since there's already a topic for [abstract syntax trees](https://github.com/topics/ast), it seems only logical to have that for concrete one's as well.

### Curating a new topic or collection

- [x] I've formatted my changes as a new folder directory, named for the topic or collection as it appears in the URL on GitHub (e.g. `https://github.com/topics/[NAME]` or `https://github.com/collections/[NAME]`)
- [ ] My folder contains a `*.png` image (if applicable) and `index.md`
- [x] All required fields in my `index.md` conform to the Style Guide and API docs: <https://github.com/github/explore/tree/main/docs>

Since there's already a topic for [abstract syntax trees](https://github.com/topics/ast), it seems only logical to have that for concrete one's as well.

### Something that does not neatly fit into the binary options above

- [ ] My suggested edits are not about an existing topic or collection, or at least not a single one
- [ ] My suggested edits are not about curating a new topic or collection, or at least not a single one
- [x] My suggested edits conform to the Style Guide and API docs: https://github.com/github/explore/tree/main/docs

---

**Please note: we will close your PR without comment if you do not check the boxes above and provide ALL requested information.**

### CODE REVIEW

File: topics/cst/index.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+created_by: 
+display_name: Concrete syntax tree
+released: 
+short_description: A concrete syntax tree is a low level representation of the parsed source in the structure defined by a grammar description.
+topic: cst
+aliases: parse-tree, parsing-tree, derivation-tree
+url:
```
Comment (underwoo16) [inline topics/cst/index.md:8]: I think we can remove these properties since they are empty:
`created_by`, `released`, `url`

**Review by underwoo16 — CHANGES_REQUESTED** [review]

File: topics/cst/index.md (hunk 1 above)
Comment (shaedrich) [reply topics/cst/index.md]: Ah, right. Did that 👍🏻

**Comment by shaedrich** [comment]
Failed check will be fixed by #3834

**Review by zkoppert — APPROVED** [review]
🎉

**Pull request merged**